assets/onboarding/data/actions.js

```diff
@@ -0,0 +1,92 @@
+import {
+	API_BASE_PATH,
+	FETCH_FROM_API,
+	FETCH_USAGE_TRACKING,
+	SUBMIT_USAGE_TRACKING,
+	ERROR_USAGE_TRACKING,
+	SET_USAGE_TRACKING,
+} from './constants';
+
+/**
+ * @typedef  {Object} FetchFromAPIAction
+ * @property {string} type               Action type.
+ * @property {Object} request            Object that is used to fetch.
+ */
+/**
+ * Fetch action creator.
+ *
+ * @param {Object} request Object that is used to fetch.
+ *
+ * @return {FetchFromAPIAction} Fetch action.
+ */
+export const fetchFromAPI = ( request ) => ( {
+	type: FETCH_FROM_API,
+	request,
+} );
+
+/**
+ * @typedef  {Object} FetchUsageTrackingAction
+ * @property {string} type                     Action type.
+ */
+/**
+ * Fetch usage tracking action creator.
+ *
+ * @return {FetchUsageTrackingAction} Fetch usage tracking action.
+ */
+export const fetchUsageTracking = () => ( {
+	type: FETCH_USAGE_TRACKING,
+} );
+
+/**
+ * Submit usage tracking action creator.
+ *
+ * @param {boolean} usageTracking Usage tracking.
+ */
+export function* submitUsageTracking( usageTracking ) {
```

Comment:
It would be better to name these like `completeWelcomeStep`, because they have other effects (creating pages in this case, advancing the step state). It also looks better for a reducer that advances the current step on COMPLETE_*_STEP.

Reply:
With the new approach I kept the `submitWelcomeStep` action creator with a similar name, but the action types are with names like:

`START_SUBMIT_SETUP_WIZARD_DATA`, `SUCCESS_SUBMIT_SETUP_WIZARD_DATA`, and `ERROR_SUBMIT_SETUP_WIZARD_DATA`.

I used `SUCCESS` instead of `COMPLETE` because when there is an error, it's also completed.